status-whisper-mailserver-spec.md

```diff
@@ -43,35 +43,35 @@ In the Whisper envelope's payload section, there MUST be RLP-encoded information
 [ Lower, Upper, Bloom, Limit, Cursor ]
 ```
 
-`Lower`: 4-byte wide unsigned integer (UNIX time in seconds; oldest requested envelope's creation time)
-`Upper`: 4-byte wide unsigned integer (UNIX time in seconds; newest requested envelope's creation time)
-`Bloom`: 64-byte wide array of Whisper topics encoded in a bloom filter to filter envelopes
-`Limit`: 4-byte wide unsigned integer limiting the number of returned envelopes
+`Lower`: 4-byte wide unsigned integer (UNIX time in seconds; oldest requested envelope's creation time)  
+`Upper`: 4-byte wide unsigned integer (UNIX time in seconds; newest requested envelope's creation time)  
+`Bloom`: 64-byte wide array of Whisper topics encoded in a bloom filter to filter envelopes  
+`Limit`: 4-byte wide unsigned integer limiting the number of returned envelopes  
 `Cursor`: an array of a cursor returned from the previous request (optional)
 
 The `Cursor` field SHOULD be filled in if a number of envelopes between `Lower` and `Upper` is greater than `Limit` so that the requester can send another request using the obtained `Cursor` value. What exactly is in the `Cursor` is up to the implementation. The requester SHOULD NOT use a `Cursor` obtained from one mailserver in a request to another mailserver because the format or the result MAY be different.
 
-The envelope MUST be signed with a symmetric key agreed between the requester and Mailserver.
+The envelope MUST be encrypted with a symmetric key agreed between the requester and Mailserver.
```

Comment:
From the code it actually looks like it is both: 
https://github.com/status-im/status-go/blob/a636f33109130a9151553abf5e0253e35994ef60/mailserver/mailserver.go#L808
Doesn't actually check the content of the signature though. Do we add it in the spec?

Reply:
This is good. When the envelope is sealed the signature is created but it's not checked.

Reply:
What is the point of requiring a signature if the signature is not checked/used in any way?

Anyhow, if we are not changing that soon, should the specification not be updated that it must be signed by whichever random key?

Reply:
It's a feature of an envelope to have the signature when encrypted. 

> Anyhow, if we are not changing that soon, should the specification not be updated that it must be signed by whichever random key?

I think the current line with your fix is the best description of the current situation.

Reply:
Feature, sure, but if I look at that validation code it looks like it requires it?

But sure, from spec point of view I'm OK with current description.

Reply:
It only requires it if a signature is provided. Signature is provided [if `Src` is defined](https://sourcegraph.com/github.com/status-im/status-go@ac5f683ac1048fe0291b0fadf75a792698114407/-/blob/whisper/message.go#L244-248).